src/jdk.compiler/share/classes/com/sun/tools/javac/code/Source.java

```diff
@@ -246,6 +246,7 @@ public enum Feature {
         RECORD_PATTERNS(JDK21, Fragments.FeatureDeconstructionPatterns, DiagKind.PLURAL),
         STRING_TEMPLATES(JDK21, Fragments.FeatureStringTemplates, DiagKind.PLURAL),
         UNNAMED_CLASSES(JDK21, Fragments.FeatureUnnamedClasses, DiagKind.PLURAL),
+        IMPLICIT_CLASSES(JDK21, Fragments.FeatureImplicitClasses, DiagKind.PLURAL),
```

Comment:
Should this be JDK22?

Reply:
No. It should be left at the first preview level.

Reply:
`UNNAMED_CLASSES` should probably be removed, as I think it is fully replaced with `IMPLICIT_CLASSES`, and I don't think there are bootstrap problems here (unlike in `jdk.internal.java.PreviewFeature`). I would say it would make sense to set the version to `JDK22`, although the difference is very small given the feature is a preview and requires `-source 22` anyway. (Although we keep the `@since` value for the duration of preview.)

Reply:
Changing

Reply:
I had the same question when reading over the changes if JDK21 or JDK22 should be used for IMPLICIT_CLASSES.

I think it would help if the comment of the Feature enum documented what the expected protocol was for previewing features.